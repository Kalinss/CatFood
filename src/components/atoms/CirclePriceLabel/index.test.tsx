import React from "react";
import { CirclePriceLabel } from "./index";
import renderer from "react-test-renderer";
import { configure, mount } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

configure({ adapter: new Adapter() });

describe("CirclePriceLabel", () => {
  it("should be rendered", () => {
    const tree = renderer
      .create(<CirclePriceLabel metric={"кг"} count="5" />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("check prop count by default", () => {
    const CircleComponent = mount(<CirclePriceLabel />);
    expect(CircleComponent.find(".count").text()).toEqual("0");
  });

  it("check prop metric by default", () => {
    const CircleComponent = mount(<CirclePriceLabel />);
    expect(CircleComponent.find(".metric").text()).toEqual("кг");
  });

  it("component should be have class", () => {
    const CircleComponent = mount(<CirclePriceLabel classStyle={"asd"} />);
    CircleComponent.debug();
    expect(CircleComponent.find(".body").hasClass("asd")).toEqual(true);
  });
});
