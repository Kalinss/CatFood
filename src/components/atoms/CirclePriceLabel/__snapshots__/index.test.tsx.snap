// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CirclePriceLabel should be rendered 1`] = `
<div
  className="body"
>
  <span
    className="count"
  >
    5
  </span>
  <span
    className="metric"
  >
    кг
  </span>
</div>
`;
